recipes/pcl/all/conanfile.py

```diff
@@ -0,0 +1,255 @@
+import os
+from glob import glob
+from itertools import chain
+
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class PclConanRecipe(ConanFile):
+    name = "pcl"
+    description = "Point Cloud Library"
+    license = "BSD-3-Clause"
+    homepage = "https://pointclouds.org/"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("pointcloud", "computer-vision", "point-cloud")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_cuda": [True, False],
+        "with_tools": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_cuda": False,
+        "with_tools": False,
+        "qhull:reentrant": False
+    }
+    requires = ("boost/1.74.0",
+                "eigen/3.3.8",
+                "flann/1.9.1",
+                "libpng/1.6.37",
+                "qhull/7.3.2")
+    generators = ["cmake", "cmake_find_package"]
+    exports = ["CMakeLists.txt"]
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _version_suffix(self):
+        semver = tools.Version(self.version)
+        return "{}.{}".format(semver.major, semver.minor)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("pcl-pcl-{}".format(self.version), self._source_subfolder)
+        cmake_lists = os.path.join(self._source_subfolder, "CMakeLists.txt")
+        tools.replace_in_file(
+            cmake_lists,
+            """set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/" ${CMAKE_MODULE_PATH})""",
+            """list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules/")"""
+        )
+        tools.replace_in_file(
+            cmake_lists,
+            "find_package(FLANN 1.7.0 REQUIRED)",
+            "find_package(Flann REQUIRED)"
+        )
```

Comment:
Not required, you should add `cmake_find_package_multi` generator in this recipe: https://github.com/PointCloudLibrary/pcl/blob/pcl-1.11.1/cmake/Modules/FindFLANN.cmake

Reply:
actually I've tested and it's required indeed, there are casing issues on Windows. It prevents find_package(FLANN) to find FindFLANN from pcl source code, breaking its logic.